verilog/analysis/flow_tree.h

```diff
@@ -0,0 +1,114 @@
+// Copyright 2017-2022 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef VERIBLE_VERILOG_FLOW_TREE_H_
+#define VERIBLE_VERILOG_FLOW_TREE_H_
+
+#include <bitset>
+#include <map>
+#include <stack>
+#include <string>
+#include <vector>
+
+#include "absl/status/status.h"
+#include "common/lexer/token_stream_adapter.h"
+#include "verilog/parser/verilog_token_enum.h"
+
+namespace verilog {
+using TokenSequenceConstIterator = verible::TokenSequence::const_iterator;
+
+// Receive a complete token sequence of one variant.
+// 1st argument: generated variant.
+// 2nd argument: number of generated variants (to decide if more is needed).
+// 3rd argument: checks if the variant is ready to be sent.
```

Comment:
Not every variant that the VariantReceiver receives is ready to be used.
Because I use the receiver callback in 2 modes:
1. To check if the user wants more variants or not (main functionality).
2. Also to terminate other DFS calls, that was started before the user says "no more" (this is where the 3rd argument will be false).

At the beginning of `FlowTree:DepthFirstSearch` I do:
```c++
  if (!receiver(current_sequence_, variants_counter_, false)) {
    return absl::OkStatus();
  }
```

Reply:
so if it is used in two modes, this needs to be clearly explained (the current third parameter description is not super helpful), but it also sounds like some unnecessary leaking of an implementation detail.

My expectation as a use of such an API is: I get called back with fully usable variants and once I return 'false', the callback will not be called anymore.